app/src/store/middleware/log.ts

```diff
@@ -0,0 +1,54 @@
+import { type Middleware } from '@reduxjs/toolkit';
+import { type Dispatch, type State } from '../app';
+
+export type LogOptions = Readonly<{
+  logStackTrace?: boolean;
+}>;
+
+export const logMiddlewareDefaultOptions: LogOptions = Object.freeze({
+  logStackTrace: false,
+});
+
+export const createLogMiddleware: (
+  options?: LogOptions,
+) => Middleware<{}, State, Dispatch> =
+  ({ logStackTrace } = logMiddlewareDefaultOptions) =>
+  () =>
+  (next) =>
+  (action) => {
+    let message: string = action.type;
+
+    if (action.type.match(/\/rejected$/g)) {
+      const error = action.payload || action.error;
+
+      const filteredError = logStackTrace
+        ? error
+        : filterOutKey('stack', error);
+
+      message += ' ' + JSON.stringify(filteredError, null, 2);
+    }
+
+    console.log(message);
+
+    return next(action);
+  };
+
+/**
+ * Recursively filter out a key from an object.
```

Comment:
would an iterative way be better?

Reply:
To be honest, probably not but I also can't think of a way to iteratively remove all keys from nested objects.